src/storage/src/table/state_table.rs

```diff
@@ -135,6 +136,22 @@ impl<S: StateStore> StateTable<S> {
             epoch,
         ))
     }
+
+    pub async fn iter_with_bounds(
+        &self,
+        pk_bounds: Range<Row>,
```

Comment:
Use `RangeBounds` in APIs (like in `StateStore`) so that we can use unbounded range e.g., `start..`?

Reply:
> btw, how to represent a prefix scan? e.g., pk=[a,b], scan a=1

`iter_with_pk_prefix`  has been added, and maybe we can keep these two interfaces(`iter_with_pk_prefix` and `iter_with_pk_bound`).

Reply:
> btw, how to represent a prefix scan? e.g., pk=[a,b], scan a=1

`iter_with_pk_prefix` has been added, and maybe we can keep these two interfaces(`iter_with_pk_prefix` and `iter_with_pk_bound`).

Reply:
Yes, but still need `RangeBounds`, which includes all of these range types
<img width="814" alt="image" src="https://user-images.githubusercontent.com/37948597/172319271-67325487-b8bc-47b3-a701-9b70d9b135ed.png">